src/gennylib/include/gennylib/quiesce.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_5936639A_7D22_4629_8FE1_4A08443DDB0F_INCLUDED
+#define HEADER_5936639A_7D22_4629_8FE1_4A08443DDB0F_INCLUDED
+
+#include <string_view>
+
+#include <mongocxx/uri.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/database.hpp>
+#include <bsoncxx/types.hpp>
+#include <bsoncxx/array/view.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/topology.hpp>
+
+namespace genny {
+
+const int DROPPED_COLLECTION_RETRIES = 1000;
+
+bool waitOplog(Topology& topology) {
+    class WaitOplogVisitor : public TopologyVisitor {
+    public:
+        void visitReplSetDescriptionPre(const ReplSetDescription& desc) {
+            using bsoncxx::builder::basic::kvp;
+            using bsoncxx::builder::basic::make_document;
+            mongocxx::client client(mongocxx::uri(desc.primaryUri));
+            auto admin = client["admin"];
+
+            // Assert that all nodes are in reasonable states.
+            auto replSetRes = admin.run_command(make_document(kvp("replSetGetStatus", 1)));
+            auto members = replSetRes.view()["members"];
+            if (members && members.type() == bsoncxx::type::k_array) {
+                bsoncxx::array::view members_view = members.get_array();
+                for (auto member : members_view) {
+                    std::string state(member["stateStr"].get_utf8().value);
+                    if (state != "PRIMARY" && state != "SECONDARY" && state != "ARBITER") {
```

Comment:
what other states could we be? standalone?

Reply:
There are a few [exotic](https://docs.mongodb.com/manual/reference/replica-states/) states if the replica set isn't in a stable position. This check was just copied from the js though we can delete if we don't think genny will ever talk to a repl set in those states.